### ISSUE TITLE
Fetch Failed with example code

### ISSUE THREAD

**Issue by HUGH-XUUUU** [ccxt/ccxt#15894]
I have use proxy in terminal and code but got this error :
ExchangeNotAvailable: okx GET https://www.okx.com/api/v5/asset/currencies  fetch failed

virsion: 
"ccxt": "^2.0.55"
"https-proxy-agent": "^5.0.1"

my code here ：

```
const ccxt = require('ccxt').pro
const  HttpsProxyAgent = require ('https-proxy-agent')
const proxy = 'https://127.0.0.1:9999'
const agent = new HttpsProxyAgent (proxy)
const exchange = new ccxt.okx({
    agent,
    enableRateLimit: true,
    apiKey: 'mykey',
    secret: 'mysecret',
    password: 'mypassword',
    options: {
        'ws': {
            'options': { agent: agent }, // ←--- httpsAgent here
        },
    },
})
exchange.setSandboxMode(true)
async function watchBalance (exchange) {
    let balance = await exchange.fetchBalance ()
    console.log ('---------------------------------------------------------')
    console.log (exchange.iso8601 (exchange.milliseconds ()))
    console.log (balance, '\n')
    while (true) {
        try {
            const update = await exchange.watchBalance ()
            balance = exchange.deep_extend (balance, update)
            // it will print the balance update when the balance changes
            // if the balance remains unchanged the exchange will not send it
            console.log ('---------------------------------------------------------')
            console.log (exchange.iso8601 (exchange.milliseconds ()))
            console.log (balance, '\n')
        } catch (e) {
            console.log (e.constructor.name, e.message)
            break
        }
    }
}
async function main() {
    console.log ('Using proxy server',agent);
    await exchange.loadMarkets()
    await watchBalance (exchange)
    await exchange.close ()
}
main()

**Comment by HUGH-XUUUU** [comment]
@znarf @xpl @blair @speedmax Help me pls!!

**Comment by frosty00** [comment]
I've fixed this #15899 

it's due to node v18 and above shipping with a different version of fetch that doesn't support httpsAgent
